app/helpers/session_helper.rb

```diff
@@ -3,16 +3,39 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   # Return the current logged-in user
   def current_user
-    @current_user ||= User.find_by id: session[:user_id] if session[:user_id]
+    if user_id = session[:user_id]
+
+      @current_user ||= User.find_by id: user_id
+    elsif user_id = cookies.signed[:user_id]
+
+      user = User.find_by id: user_id
+      if user&.authenticated?(cookies[:remember_token])
```

Comment:
user&.authenticated?
The Safe Navigation Operator (&.) để làm gì em ?

Reply:
đây là viết tắt của user & user.authenticated . cách viết này để không đưa ra exception khi user =nil và không truy cập dc authenticated